src/components/SignupForm.tsx

```diff
@@ -0,0 +1,124 @@
+import { ChangeEvent, useRef } from "preact/compat";
+import { User, createUser } from "../api/createUser";
+
+const SignupForm = () => {
+  const formRef = useRef<HTMLFormElement>(null);
+
+  const handleCreateUser = async (e: ChangeEvent) => {
+    e.preventDefault();
+    try {
+      if (formRef.current) {
+        const formData = new FormData(formRef.current);
+        const { first_name, last_name, email, password } = Object.fromEntries(
+          formData.entries()
+        ) as User;
+        const newUser = await createUser({
+          first_name,
+          last_name,
+          email,
+          password,
+        });
+        console.log(newUser);
+        console.log("user created");
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  };
+  return (
+    <form class="space-y-6" onSubmit={handleCreateUser} ref={formRef}>
+      <div class="flex justify-between items-center">
+        <div>
+          <label
```

Comment:
make these into components like did in SES template manager.
Always follow DRY and modular code.